### PR TITLE
define schema for dolt_schemas table

### PR DESCRIPTION
**Pull request by jennifersp**
https://github.com/dolthub/doltgresql/pull/454 depends on this PR.

### CODE REVIEW

**Review by zachmu — COMMENTED** [review]
I don't think this is the right approach.

The dolt_schemas table defines elements like views and triggers for tables in the database. It follows that we should have a dolt_schemas table in every database schema, to define the views etc. for that schema.

This is required to address e.g. https://github.com/dolthub/doltgresql/issues/456

So this method needs to change: it need to specify a schema string (empty in the case of dolt).

**Comment by coffeegoddd** [comment]
@jennifersp __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>0f6d6cc</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>0f6d6cc</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@jennifersp __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>e6f9398</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>e6f9398</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/libraries/doltcore/sqle/schema_table.go [hunk 1]
```diff
@@ -69,6 +70,20 @@ var schemaTableSchema = schema.MustSchemaFromCols(schemasTableCols)
...
+	}
+
+	schemaName := doltdb.DefaultSchemaName
+	if resolve.UseSearchPath && db.schemaName == "" {
+		schemaName, err = resolve.FirstExistingSchemaOnSearchPath(ctx, root)
+		if err != nil {
+			return nil, err
+		}
+		db.schemaName = schemaName
+	}
```
Comment (Hydrocharged) [inline go/libraries/doltcore/sqle/schema_table.go:85]: So if `db.schemeName` has a value (such as `example_schema`), then we're never setting that schema? That's what it looks like from this section at least. We start with `doltdb.DefaultSchemaName`, and then resolve the schema name only if `db.schemaName == ""`. So if `db.schemaName != ""`, then we never assign and use it.

**Review by Hydrocharged — APPROVED** [review]
Tests pass, but the part highlighted in the comment doesn't seem quite right.

**Comment by coffeegoddd** [comment]
@jennifersp __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>7a84e8c</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>7a84e8c</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**